test/unit/BPool/BPool_Bind.tree

```diff
@@ -0,0 +1,26 @@
+BPool::Bind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
```

Comment:
I think it's related to this: https://github.com/defi-wonderland/balancer-v1-amm/pull/129#discussion_r1664994114

if more of the expected preconditions are met by default (ie: by having the test contract be the deployer, and therefore controller, of the pool) then it's easier to flatten the tree since the non-reverting case is free and the caller _not_ being the controller requires some setup.

While I could give this a go, I think in order give branching tree technique a proper shot we should be a bit less afraid of our trees actually having branches (?

Reply:
keep in mind that if we have one tree file per function then it's less likely that two modifiers with the same name require a different implementation as well

Reply:
> just a comment here, aimed to try to get some consistency around our trees.
> 
> i find that `when caller is controller` is redundant, since the branch above is `when caller is NOT controller`, then we must assume that what happens after that branch is indeed called by the controller
> 
> now, the issue is that, if we do use `when caller is controller`, we should extend it to `when reentrancy lock is NOT set and caller is controller`, and if we don't, we need to use a common text such as `when it fulfils general conditions` or sth like that
> 
> i'd incline myself for the 2nd, having a generic "when it doesn't fail", but then we do have a lot of `when`s that it actually fails inside, so i wouldn't know how to name this branch

Why not both actually? if it's just one condition (caller is controller/not controller), then having both explicitly expressed as nodes is clear, while if it's a X conditions setup, a 'when all conditions are met' or similar makes more sense?

Reply:
or like ln21, "when token can be bound" is pretty good at carrying the branch info too imo

Reply:
-> biggest use-case of trees isn't to be updated tho, but rather to give an overview of the unit test which is super easy to read/grasp (internal review/audits) and have a systematic approach aimed at challenging every branch of the unit tested

Reply:
> biggest use-case of trees isn't to be updated tho

but without the CI checker, the overview of the unit test can be misleading, as the reader may be expecting that some things are being checked but they're actually not even implemented.

anyways, without the comments being also checked, someone could indeed delete the comment and skip the expectation, but at least is a bigger misconduct than just "forgetting to implement a whole test"